Packs/TOPdesk/Integrations/TOPdesk/TOPdesk.py

```diff
@@ -1272,6 +1295,226 @@ def fetch_incidents(client: Client,
+    if parsed_args.incident_changed:
+        demisto.debug(f'Incident changed: {parsed_args.incident_changed}')
+        # Close ticket if needed
+        if parsed_args.inc_status == IncidentStatus.DONE and params.get('close_ticket'):
+            # Set status TOPdesk ticket to Closed
+            demisto.debug('Close TOPdesk ticket')
+            # Close with API call or set field and let mirroring handle it.
+            # client.update_incident
+        # TODO: Something with updated delta keys or not?
+        # 'processingStatus', 'priority', 'urgency', 'impact'
+        update_args = {
+            'id': ticket_id
+        }
+        for key in parsed_args.delta:
+            update_args[key] = parsed_args.delta[key]
+
+        demisto.debug(f'SZU update_args=[{update_args}]')
+        client.update_incident(update_args)
+
+    entries = parsed_args.entries
+    if entries:
+        demisto.debug(f'New entries {entries}')
+
+        for entry in entries:
+            demisto.debug(f'Sending entry {entry.get("id")}, type: {entry.get("type")}')
+            # Mirroring files as entries
+            if entry.get('type') == 3:
+                path_res = demisto.getFilePath(entry.get('id'))
+                full_file_name = path_res.get('name')
+                file_name, file_extension = os.path.splitext(full_file_name)
+                if not file_extension:
+                    file_extension = ''
+                client.attachment_upload(incident_id=ticket_id, incident_number=None, file_entry=entry.get('id'),
+                                         file_name=file_name + '_mirrored_from_xsoar' + file_extension,
+                                         invisible_for_caller=False,
+                                         file_description=f"Upload from xsoar: {file_name}.{file_extension}")
+            else:
+                # Mirroring comment and work notes as entries
+                xargs = {
+                    'id': '',
+                    'action': '',
+                    'action_invisible_for_caller': False,
+                }
+                tags = entry.get('tags', [])
+                if params.get('work_notes_tag') in tags:
+                    xargs['action_invisible_for_caller'] = True
+                elif params.get('comments_tag') in tags:
+                    xargs['action_invisible_for_caller'] = False
+                # Sometimes user is an empty str, not None, therefore nothing is displayed
+                user = entry.get('user', 'dbot')
+                if (user):
+                    duser = demisto.findUser(username=user)
+                    name = duser['name']
+                else:
+                    name = 'Xsoar dbot'
+
+                text = f"<i><u>Update from {name}:</u></i><br><br>{str(entry.get('contents', ''))}" \
+                    + "<br><br><i>Mirrored from Cortex XSOAR</i>"
+                # client.add_comment(ticket_id, ticket_type, key, text)
```

Comment:
Does the text need to be sent this way?
Why is the `add_comment` call commented?

Reply:
I removed the test rule (client.add_comment)

The text is formatted this way, so the user in TOPdesk could see who did the update in Xsoar and the message has been mirrored from Xsoar (requirement from our team).

Is it OK to leave it in this format?

Reply:
Yes